architecture/2024-05-radius-on-dapr.md

```diff
@@ -0,0 +1,324 @@
+# Radius on Dapr
+
+* **Status**: Pending/Approved
+* **Author**: Ryan Nowak (`@rynowak`)
+
+## Overview
+
+The design doc proposes replacing some existing plumbing in Radius with [Dapr](https://dapr.io) and taking an install-time dependency on Dapr. Radius needs a database and a message queue to do its work. Dapr provides implementations of its [building block APIs](https://docs.dapr.io/concepts/building-blocks-concept/) that work across self-hostable open-source technology and managed cloud services. Compared to developing our own data layer, adopting Dapr will accellerate development and production-readiness of Radius because Dapr is already mature and production-hardened. Dapr provides SDKs for most languages, including Go and C# so it can be used in the Deployment Engine.
```

Comment:
We spent time last year removing Dapr from installer due to download speed. Is this still a concern?

Reply:
Yes. The installation time is a drawback. We're probably talking about adding a minute to install time for local kubernetes, and much less for cloud-hosted.